### ISSUE TITLE
Java 9+ Illegal reflective access on java.net.URI

### ISSUE THREAD

**Issue by eggilbert** [jqno/equalsverifier#214]
This looks to be the same as: https://github.com/jqno/equalsverifier/issues/207, https://github.com/jqno/equalsverifier/issues/193, https://github.com/jqno/equalsverifier/issues/183, https://github.com/jqno/equalsverifier/issues/172

# What steps will reproduce the problem?

Use EqualsVerifier on a class containing `java.net.URI`.

# What is the code that triggers this problem?

```java
package my.company;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.junit.jupiter.api.Test;

import java.net.URI;
import java.util.Objects;

class URITest {
    static class POJO {
        private URI field;

        public URI getField() {
            return field;
        }

        public void setField(URI field) {
            this.field = field;
        }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (o == null || getClass() != o.getClass()) return false;
            POJO pojo = (POJO) o;
            return Objects.equals(field, pojo.field);
        }

        @Override
        public int hashCode() {
            return Objects.hash(field);
        }
    }

    @Test
    void equalsContract() {
        EqualsVerifier.forClass(POJO.class)
            .usingGetClass()
            .suppress(Warning.NONFINAL_FIELDS)
            .verify();
    }
}
```

# What error message or stack trace does EqualsVerifier give?

```
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.scheme
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.fragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.authority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.userInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.host
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.port
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.path
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.query
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.schemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.hash
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedUserInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedAuthority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedPath
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedQuery
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedFragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedSchemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.string
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.scheme
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.fragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.authority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.userInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.host
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.port
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.path
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.query
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.schemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.hash
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedUserInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedAuthority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedPath
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedQuery
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedFragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedSchemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.string
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.scheme
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.fragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.authority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.userInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.host
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.port
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.path
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.query
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.schemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.hash
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedUserInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedAuthority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedPath
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedQuery
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedFragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedSchemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.string
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.scheme
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.fragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.authority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.userInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.host
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.port
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.path
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.query
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.schemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.hash
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedUserInfo
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedAuthority
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedPath
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedQuery
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedFragment
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.decodedSchemeSpecificPart
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (...) to field java.net.URI.string
```

# What did you expect?

No compiler warnings emitted

# Which version of EqualsVerifier are you using?

3.0

# Please provide any additional information below.

Discovered with JDK 10.0.1

**Comment by jqno** [comment]
Thanks for reporting this! I've added a prefab value for `java.net.URI` which fixes this issue. I'll let you know when I make a release.

In the mean time, as a workaround, you can add prefab values for `URI` yourself, like this:

```
EqualsVerifier.forClass(...)
        .withPrefabValues(URI.class, URI.create("x"), URI.create("y"))
        .verify();
```

**Comment by eggilbert** [comment]
@jqno interestingly, when I add `.withPrefabValues(URI.class, URI.create("x"), URI.create("y"))` to my test case above I still receive reflection warnings, but far less of them. It goes from 68 warnings down to 17 (looks to be 1 copy of each instead of 4).

🤔

**Comment by jqno** [comment]
That's strange. Perhaps there's another class, next to `URI`, that gives the same warnings. If you let me know which one, I can add prefab values for that as well.

It will be a while before I can respond next though, since I'm going out of town.

**Comment by eggilbert** [comment]
It looks like the remaining warnings are triggered by making the "redCopy" in `PrefabValuesApi#addPrefabValues`. 

I built the latest snapshot version with your change, and there aren't any reflection warnings if I remove that `withPrefabValues()` call though.

**Comment by jqno** [comment]
Ah, that makes sense. Thanks for checking.

If this is an urgent issue for you, I'll try to make a release soon. Otherwise I'll try to squeeze in some other things before I do a release.

**Comment by eggilbert** [comment]
It's not urgent, no. But this is currently the only reflection warning I have left to mitigate, so sometime in the next few weeks would be nice just to tie up loose ends 😺

**Comment by jqno** [comment]
OK, I'll try to make a release soon, then. I'll let you know when I do.

In the mean time, I've also opened #215 to address the more general case where `withPrefabValues` can still generate illegal reflective access warnings.

**Comment by jqno** [comment]
I've released version 3.0.1, which should fix this issue.
